pkg/nb/rpc.go

```diff
@@ -2,28 +2,56 @@ package nb
 
 import (
 	"bytes"
+	"context"
 	"crypto/tls"
+	"encoding/binary"
 	"encoding/json"
+	"fmt"
+	"io"
 	"io/ioutil"
 	"net/http"
 	"strings"
-	"time"
+	"sync"
 
+	"github.com/noobaa/noobaa-operator/v2/pkg/util"
+	"github.com/prometheus/common/log"
 	"github.com/sirupsen/logrus"
+	"nhooyr.io/websocket"
 )
 
+// RPCVersionNumber specifies the RPC version number
+const RPCVersionNumber uint32 = 0xba000000
+
+// GlobalRPC is the global rpc
+var GlobalRPC *RPC
+
+func init() {
+	// GlobalRPC initialization
+	GlobalRPC = NewRPC()
+}
+
+// RPC is a struct that describes the relevant fields upon handeling rpc protocol
+type RPC struct {
+	Protocol    string
```

Comment:
I think Protocol should be a property of the router, but do we even need it now?